src/split-pane/index.ts

```diff
@@ -111,28 +114,59 @@ export class SplitPaneBase<P extends SplitPaneProperties = SplitPaneProperties>
 
 	protected getPaneStyles(): {[key: string]: string} {
 		const {
-			direction = Direction.row,
+			direction = Direction.column,
 			size = DEFAULT_SIZE
 		} = this.properties;
+
 		const styles: {[key: string]: string} = {};
 
-		styles[direction === Direction.row ? 'width' : 'height'] = `${size}px`;
+		let computedSize: string | number = this._collapsed ? 'auto' : size;
+
+		styles[direction === Direction.column ? 'width' : 'height'] = `${computedSize}px`;
+
 		return styles;
 	}
 
+	protected onAttach() {
+		this._collapseIfNecessary();
+	}
+
+	private _collapseIfNecessary = () => {
```

Comment:
@tomdye Do you think we should react to a change to the `collapseWidth` via properties and set the `this._collapsed` flag and call `onCollapse` appropriately? Otherwise if a different `collapseWidth` is passed the component won't react until it is is resized?

Reply:
Guess the same applies if the `Direction` is change also...